### DISCUSSION TITLE
Initial component best practice

### DISCUSSION THREAD

**Discussion by eheck-is** [WeblateOrg/weblate#12240] [Q&A]
From what I gather from the documentation and the UI there is no way to "attach" a repository to a project (or something else) instead of a component. Once I have a component "comp1" in "proj1" I can access the repository from other components with "weblate://proj1/comp1" as the repository string. But what if the file referenced in the first component gets deleted, especially when using the discovery extension?

I did an experiment with a setup like this in a test project and then deleted the initial component. The result was an error message about a constraint violation in the database and afterwards all the components were gone.

How would I handle this? Set up my initial component with some dummy file that must never be deleted? Sounds ugly to me, but what is the alternative?

**Comment by nijel** [comment]
The initial component should be something that lasts forever.

**Comment by eheck-is** [comment]
I get that. But how do I achieve that? Do I really have to create a dummy file, because that will be the only thing that is guaranteed to not be deleted or moved?

**Reply by nijel** [reply]
It really depends on your project. Often there is some core library or index document that is very unlikely to disappear.
